core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -95,6 +97,20 @@ public int getBusyWriterCount() {
         return writerPool.getBusyCount();
     }
 
+    public void migrateNullFlag(TableWriter writer, CairoSecurityContext cairoSecurityContext, CharSequence tok) {
+        TableReader reader = getReader(cairoSecurityContext, tok);
+        TableReaderMetadata readerMetadata = (TableReaderMetadata) reader.getMetadata();
+        if (readerMetadata.getVersion() < 416){
+            for (int i = 0; i < reader.getColumnCount(); i++) {
+                if (readerMetadata.getColumnType(i) == SYMBOL) {
+                    if (reader.hasNull(i)) {
+                        writer.getSymbolMapWriter(i).updateNullFlag();
```

Comment:
table metadata should also be updated, so that distinct uses better algo after update